test/e2etest/crowd_test.go

```diff
@@ -0,0 +1,262 @@
+	assert.Nil(t, kubectlError)
+
+	// delete postgres-client pod otherwise Terraform will fail to delete the namespace
+	log.Print("Deleting postgres-client pod")
+	_, kubectlError = k8s.RunKubectlAndGetOutputE(t, kubectlOptions,
+		"delete", "pod", "postgres-client", "-n", "atlassian", "--force", "--grace-period=0")
+	assert.Nil(t, kubectlError)
+
+	// copy processed crowd.cfg.xml file to crowd shared home
+	log.Print("Copying crowd.cfg.xml to crowd pod")
+	_, kubectlError = k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "cp", "crowd.cfg.xml", "atlassian/crowd-0:/var/atlassian/application-data/crowd/shared/crowd.cfg.xml", "-n", "atlassian")
+	assert.Nil(t, kubectlError)
+
+	// rollout restart crowd StatefulSet so that the copied crowd.cfg.xml is found
+	log.Print("Restarting crowd pod")
+	_, kubectlError = k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "rollout", "restart", "sts/crowd", "-n", "atlassian")
+	assert.Nil(t, kubectlError)
+
+	// wait for crowd deployment to pass readiness probe
+	log.Print("Waiting for crowd pod to be ready")
+	_, kubectlError = k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "rollout", "status", "sts/crowd", "-n", "atlassian", "--timeout=300s")
+	assert.Nil(t, kubectlError)
+
+	// if calls to crowd are made right away, occasional 503 errors are possible
+	// giving Crowd extra 10 seconds fixes it, however, looking at the correctness
+	// of crowd readiness probe is a better idea
+	time.Sleep(10 * time.Second)
+}
+
+func crowdTests(t *testing.T, testConfig TestConfig, bitbucketURL string, crowdURL string, jdbcURL string) {
+	printTestBanner(crowd, "Tests")
+
+	restoreCrowdFromBackup(t, testConfig, jdbcURL)
+
+	userNameBase := "user"
+	userName := fmt.Sprintf("%s-%s", userNameBase, testConfig.EnvironmentName)
+	createNewCrowdUser(t, userName, crowdURL, testConfig.CrowdPassword)
+
+	// before making calls to Bitbucket make sure we land on the same node and avoid using sticky cookie in requests
+	// scale bitbucket to 1 replica instead of 3
+	log.Print("Scaling Bitbucket to 1")
+	_, kubectlError := k8s.RunKubectlAndGetOutputE(t, getKubectlOptions(t, testConfig), "scale", "sts/bitbucket", "-n", "atlassian", "--replicas=1")
+	assert.Nil(t, kubectlError)
+
+	// get BITBUCKETSESSIONID to use in the header in subsequent calls
+	// even though basic auth works, atl_token is different each time
+	bitbucketSessionID := getBitbucketSessionID(bitbucketURL, "admin", testConfig.BitbucketPassword)
+	assert.NotEmptyf(t, bitbucketSessionID, "BITBUCKETSESSIONID cannot be empty")
+
+	// now we need to extract atl_token from the hidden input in HTML response
+	atlToken := getAtlToken(t, bitbucketURL, bitbucketSessionID)
+	assert.NotEmptyf(t, atlToken, "atl_token cannot be empty")
+
+	// add a new user directory in Bitbucket
+	addCrowdUserDirectory(t, testConfig.EnvironmentName, bitbucketURL, crowdURL, testConfig.CrowdPassword, bitbucketSessionID, atlToken)
+	// even though sync interval is set to 1 minute, it roughly takes 2-3 mins
```

Comment:
There's no REST API to add and sync user directories, and syncing user directory will require getting its ID which is returned in html. Just waiting for a scheduled sync is simpler.